department-of-veterans-affairs/vets-website
# Dbex/84062 claim certification

Are you removing, renaming or moving a folder in this PR?

Did you change site-wide styles, platform utilities or other infrastructure?

## Summary
- Adds the claim certification component to the Review page
- Note: Changes for this feature are all behind environment flags since we are also making changes to the Intro page, which cannot be gated by a toggle. As such, e2e tests will be in a weird state until launch (staging review currently scheduled for 12/17/24). Notes were added in `cypress.helpers.js` for a temporary workaround on local environments

team: @department-of-veterans-affairs/dbex-trex 
toggle cleanup: https://github.com/department-of-veterans-affairs/va.gov-team/issues/96383

## Related issue(s)
department-of-veterans-affairs/va.gov-team#84062


## Testing done
- Manual and e2e tests. Verified unit tests are still successful (regression)

**Manual test setup**
1. Currently this feature only runs on a local dev environment
2. Sign in as a test user. Start a new claim, filling out all required fields
3. Once you get to the Review page, **verify** the new certification component displays as expected
4. Leave both the 'Your full name' and certification checkbox fields empty. **Verify** an error is displayed for the missing name and missing checkbox.
5. Leave either the name or checkbox empty. **Verify** appropriate error is displayed
6. Fill out both name and checkbox. Verify you are able to submit the claim
fyi: the platform logic that verifies the entered name matches what is in the Veteran's profile strips out whitespace and uses lowercase for the comparison.

## Screenshots
|desktop|mobile|
| ------------- | ------------- |
|![desktop](https://github.com/user-attachments/assets/4544fc4c-f7a8-4ed1-8ea1-8c45083e7098)|![mobile](https://github.com/user-attachments/assets/4b5f64f7-b2fb-459a-9069-1fab41ebd832)|

**Error example**
![claim-certification-error-example](https://github.com/user-attachments/assets/b5164f93-99d4-40cc-871d-409414cdfd1f)

## What areas of the site does it impact?
526ez

## Acceptance criteria

### Quality Assurance & Testing


### Error Handling


### Authentication


## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_
